src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -580,20 +595,22 @@ public function viewAction(int $orderId, Request $request): Response
      * @return RedirectResponse
      */
     #[AdminSecurity("is_granted('update', request.get('_legacy_controller')) && is_granted('delete', request.get('_legacy_controller'))", redirectRoute: 'admin_orders_view', redirectQueryParamsToKeep: ['orderId'], message: 'You do not have permission to edit this.')]
-    public function partialRefundAction(int $orderId, Request $request)
-    {
-        $formBuilder = $this->get('prestashop.core.form.identifiable_object.builder.cancel_product_form_builder');
-        $formHandler = $this->get('prestashop.core.form.identifiable_object.partial_refund_form_handler');
+    public function partialRefundAction(
+        int $orderId,
+        Request $request,
+        #[Autowire(service: 'prestashop.core.form.identifiable_object.builder.cancel_product_form_builder')] FormBuilder $formBuilder,
+        #[Autowire(service: 'prestashop.core.form.identifiable_object.partial_refund_form_handler')] FormHandlerInterface $formHandler,
+    ) {
         $form = $formBuilder->getFormFor($orderId);
 
         try {
             $form->handleRequest($request);
             $result = $formHandler->handleFor($orderId, $form);
             if ($result->isSubmitted()) {
                 if ($result->isValid()) {
-                    $this->addFlash('success', $this->trans('A partial refund was successfully created.', 'Admin.Orderscustomers.Notification'));
+                    $this->addFlash('success', $this->trans('A partial refund was successfully created.', [], 'Admin.Orderscustomers.Notification'));
                 } else {
-                    $this->addFlashFormErrors($form);
+                    $this->addFlashErrors(explode("\n", (string) $form->getErrors(true)));
```

Comment:
I'm not sure this will have an equivalent behaviour, maybe this method `addFlashFormErrors` should be added in `PrestaShopAdminController` This string casting with explode based on new line feels too much like a hack